segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerHealthContributor.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.segmentstore.server.host.health;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.Service;
+import io.pravega.segmentstore.server.SegmentContainer;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.Status;
+import io.pravega.shared.health.impl.AbstractHealthContributor;
+import lombok.NonNull;
+
+
+/**
+ *  A contributor to manage health of segment container.
+ */
+public class SegmentContainerHealthContributor extends AbstractHealthContributor {
+    private final SegmentContainer segmentContainer;
+
+    public SegmentContainerHealthContributor(@NonNull SegmentContainer segmentContainer) {
+        super("SegmentContainer");
+        this.segmentContainer = segmentContainer;
+    }
+
+    @Override
+    public Status doHealthCheck(Health.HealthBuilder builder) {
+        Status status = Status.DOWN;
+
+        boolean newState = segmentContainer.state() == Service.State.NEW;
```

Comment:
Why you create a new `boolean` variable every time? Wouldn't it be equivalent to do something like:
```
if (segmentContainer.state() == Service.State.NEW) {
    status = Status.NEW;
} else if (segmentContainer.state() == Service.State.STARTING) {
    status = Status.STARTING;
} else if (segmentContainer.state() == Service.State.RUNNING) {
    status = Status.UP;
}
```